### PR TITLE
Automated cherry pick of #91690: fieldManager: Ignore and log all errors when updating

### PR DESCRIPTION
**Pull request by apelisse**
Cherry pick of #91690 on release-1.16.

#91690: fieldManager: Ignore and log all errors when updating

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```
Resolve regression in metadata.managedFields handling in create/update/patch requests not using server-side apply
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 				return
 			}
 
-			obj, err = scope.FieldManager.Update(liveObj, obj, managerOrUserAgent(options.FieldManager, req.UserAgent()))
+			obj = scope.FieldManager.UpdateNoErrors(liveObj, obj, managerOrUserAgent(options.FieldManager, req.UserAgent()))
 			if err != nil {
 				scope.err(fmt.Errorf("failed to update object (Create for %v) managed fields: %v", scope.Kind, err), w, req)
 				return
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go:159]: shouldn't these error-handling lines be dropped?

**Comment by liggitt** [comment]
one comment, then lgtm

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go (hunk 1 above)
Comment (apelisse) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go:159]: Yeah, that's clearly wrong, thanks :-)

**Pull request merged**